src/script/sign.cpp

```diff
@@ -187,6 +187,11 @@ static bool SignTaproot(const SigningProvider& provider, const BaseSignatureCrea
         sigdata.tr_spenddata.Merge(spenddata);
     }
 
+    // Detect if this is the dummy signer. If so, we want to choose the largest stack size for worst case size estimation
```

Comment:
That's pretty ugly. What about adding a "bool ForFeeEstimation()" to all SignatureCreator instances, and calling that instead?

Reply:
I added an `IsDummy` and changed to check that instead.